app/assets/javascripts/Components/graders_manager.jsx

```diff
@@ -79,20 +103,18 @@ class GradersManager extends React.Component {
     }).then(this.fetchData);
   };
 
+  assignedGraders = () => {
+    return this.gradersTable.state.selection;
+  };
+
   assignRandomly = () => {
     let groups = this.groupsTable ? this.groupsTable.state.selection : [];
     let criteria = this.criteriaTable ? this.criteriaTable.state.selection : [];
-    let graders = this.gradersTable.state.selection;
-
-    if (groups.length === 0 && criteria.length === 0) {
-      alert(I18n.t("groups.select_a_group"));
-      return;
-    }
-
-    if (graders.length === 0) {
-      alert(I18n.t("graders.select_a_grader"));
-      return;
-    }
+    let graders = this.assignedGraders();
+    this.setState({
```

Comment:
a separate `setState` call here isn't necessary; you can reset this state in the implementation of `fetchData`